#! /usr/bin/env python

PACKAGE='tara_camera_driver'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       	  Type      Level Description     Default Min   Max
gen.add("exposure",       int_t,    0,    "Manual exposure", 1000,     10, 1000000)
gen.add("brightness",       int_t,    0,    "Manual brightness", 4,     1, 7)
gen.add("autoExposure",       bool_t,    0,    "Auto exposure on", True)
gen.add("targetBrightness",       int_t,    0,    "Desired brightness", 128,     0, 255)
gen.add("exposureGain",       double_t,    0,    "Exposure control gain", 1,     0, 5)

exit(gen.generate(PACKAGE, "tara_camera_driver", "taraCamera"))

